src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -305,6 +329,56 @@ class ClusterObserver : IDisposable
             return false;
         }
 
+        private const DatabaseStatus BadFlags = Faulted | Shutdown;
+        private bool ShouldReassign(
+            DatabaseTopology topology, string node, string db,
+            Dictionary<string, ClusterNodeStatusReport> current, 
+            Dictionary<string, ClusterNodeStatusReport> previous)
+        {
+            if (topology.DynamicNodesDistribution == false)
+                return false;
+
+            DatabaseStatusReport currentDbStats = null;
+            DatabaseStatusReport prevDbStats = null;
+            var hasCurrent = current.TryGetValue(node, out var currentNodeStats) && currentNodeStats.LastReport.TryGetValue(db, out currentDbStats);
+            var hasPrev = previous.TryGetValue(node, out var prevNodeStats) && prevNodeStats.LastReport.TryGetValue(db, out prevDbStats);
+
+            if (hasCurrent == false && hasPrev == false)
+                return true;
+
+            // Wait until we have more info
+            if (hasCurrent == false || hasPrev == false)
+                return false;
+
+            if (currentNodeStats.LastReportStatus != ClusterNodeStatusReport.ReportStatus.Ok &&
```

Comment:
Need to take timing information into this. We don't immediately switch at the drop of a hat. We need to only switch when enough time (configurable) have passed.